### PR TITLE
29-profile-page

### PR DESCRIPTION
**Pull request by oijbaker**
Added profile page, the settings tile is not implemented as this is a different task

### CODE REVIEW

**Review by hiba-aldalaty — COMMENTED** [review]
Couple of issues:
Design wise, it should look more like this: 
image
- There's no bio section at the moment
- Top 3 communities 
- Margin between text needs to be smaller in terms of the name, username, leaderboard position etc.. just to match the design a bit better
- Buttons in the top right corner

Functionality wise, 
- For some reason I'm, getting this console error

image

And this is what shows up instead of the graph: 
image

**Comment by hiba-aldalaty** [comment]
Still getting that error in the console? Are you getting it too?
Also, the page should not overflow horizontally

**Comment by hiba-aldalaty** [comment]
There are a couple of console errors, if you check the errors please make sure they're all fixed and no console errors come up. Some otherones are also created when in edit mode

**Review by WilliamOrringe — CHANGES_REQUESTED** [review]
Quite a few styling issues, also if editing profile can't get back to profile.
here is a list of things:
> Styling of input boxes and positioning
> Making sure that the data does not go off the page and so the navbar doesn't expand
>Navbar isn't locked to bottom for some reason
> Make the sections styled to be in line with the design: i.e. remove underlines.
add spacing in-between points and position change colour of the icons.
> add edit directly for bio under bio.
>you seem to be getting user 1s data not the currently logged in user.
> Also graph is bugging

I will help you with this task as this seems like a very big task overall and should probably been split into different tasks. I will message you and we can do some pair programming to sort these if you want to. I think it looks like it's all there just issues with styling and following the design.

File: frontend/src/views/Profile.vue [hunk 1]
```diff
@@ -1,15 +1,303 @@
...
+              this.communities.push(
+                  response.data.graphs.top_communities[i].community
+              );
+            }
+            this.initials =
+                this.firstName.substring(0, 1) + this.secondName.substring(0, 1);
+            this.bio = response.data.descrtiption;
+          })
+        .catch((error) => {
+          console.log(error);
```
Comment (sorinburghiu2323) [inline frontend/src/views/Profile.vue:137]: NIT: Should be `console.error`

File: frontend/src/views/Profile.vue [hunk 2]
```diff
@@ -1,15 +1,303 @@
...
+        });
+    },
+  },
+  data() {
```
Comment (sorinburghiu2323) [inline frontend/src/views/Profile.vue:141]: NIT: `data()` really should be at the top of the script by vue convensions.

File: frontend/src/views/Profile.vue [hunk 3]
```diff
@@ -1,15 +1,303 @@
...
+      secondName: "",
+      isTeacher: false,
+      points: 0,
+      initials: "",
+      leaderboardPosition: 0,
+      leaderboardInfo: [],
+      communities: [],
+      bio: "",
+      displayGraph: true,
+      labels: ["This week", "Last week", "2 Weeks Ago", "3 Weeks Ago"],
```
Comment (sorinburghiu2323) [inline frontend/src/views/Profile.vue:155]: You could compile all of these fields into one object. Then you dont have to use `this.first_name`, `this.last_name`, etc. You could just use `this.user = response.data.user`

Comment (WilliamOrringe) [reply frontend/src/views/Profile.vue:155]: honestly these dont impact functionallity so this should get merged as we need this for other things and we can change them on another branch

Comment (sorinburghiu2323) [reply frontend/src/views/Profile.vue:155]: I am aware. Its just a NIT

**Pull request merged**